#!/usr/bin/env python

PACKAGE = "auv_motor_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("setpoint_surge", double_t, 0, "setpoint_surge", 0, -500, 500)
gen.add("setpoint_sway", double_t, 0, "setpoint_sway", 0, -500, 500)
gen.add("setpoint_heave", double_t, 0, "setpoint_heave", 0, -500, 500)
gen.add("setpoint_yaw", double_t, 0, "setpoint_yaw", 0, -500, 500)

gen.add("plantState_surge", double_t, 0, "plantState_surge", 0, -500, 500)
gen.add("plantState_sway", double_t, 0, "plantState_sway", 0, -500, 500)
gen.add("plantState_heave", double_t, 0, "plantState_heave", 0, -500, 500)
gen.add("plantState_yaw", double_t, 0, "plantState_yaw", 0, -500, 500)

gen.add("startSwitch", bool_t, 0, "startSwitch", False)
gen.add("killSwitch", bool_t, 0, "killSwitch", False)


exit(gen.generate(PACKAGE, "auv_motor_control", "pushValues"))
